x-pack/plugins/serverless_search/public/application/components/api_key/create_api_key_flyout.tsx

```diff
@@ -82,8 +103,14 @@ export const CreateApiKeyFlyout: React.FC<CreateApiKeyFlyoutProps> = ({
   const { euiTheme } = useEuiTheme();
   const [name, setName] = useState('');
   const [expires, setExpires] = useState<string | null>(DEFAULT_EXPIRES_VALUE);
+  const [isShowingReadOnlyBoilerplate, setShowReadOnlyBoilerplate] = useState<boolean | undefined>(
+    undefined
+  );
+  const [readOnlyBoilerplate, setReadOnlyBoilerplate] = useState<string>(READ_ONLY_BOILERPLATE);
```

Comment:
I'm not following why we need state values for these. Why can we not just call `onChangeRoleDescriptors()` with the constant value when the user clicks the button?

Reply:
Added this change,  if we want to update and save updated boilerplate code?

Reply:
You should be able to change the contents after loading the value. I'll make a comment with what I'm suggesting.